### PR TITLE
fix(IAM Policy Management): add restore policy functionality

### PR DESCRIPTION
**Pull request by swcolley**
## PR summary

Updating SDK to include new changes from API for PATCH for v1/policies and additional new params.

## PR Checklist
Please make sure that your PR fulfills the following requirements:  
- [x] The commit message follows the [Angular Commit Message Guidelines](https://github.com/angular/angular/blob/master/CONTRIBUTING.md#-commit-message-guidelines).
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## Does this PR introduce a breaking change?    
- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by padamstx** [comment]
If this PR is ready for review, please sync it up with the "main" branch and remove the "draft" tag.

File: iampolicymanagementv1/iam_policy_management_v1_integration_test.go [hunk 1]
```diff
@@ -235,6 +235,32 @@ var _ = Describe("IAM Policy Management - Integration Tests", func() {
...
+	Describe("Patch an access policy", func() {
+
+		It("Successfully patched an access policy", func() {
+			shouldSkipTest()
+			Expect(testPolicyId).To(Not(BeNil()))
+			Expect(testPolicyETag).To(Not(BeNil()))
+
+			// Construct an instance of the PatchPolicyOptions model
+			options := new(iampolicymanagementv1.PatchPolicyOptions)
+			options.PolicyID = core.StringPtr(testPolicyId)
```
Comment (padamstx) [inline iampolicymanagementv1/iam_policy_management_v1_integration_test.go:252]: This is fine the way you have it, but the core XXXPtr() functions are really only needed when dealing with literal values.  For example, in Go you can't use something like &"some string" to take the address of the literal string, hence the need for the XXXPtr() functions.
```suggestion
			options.PolicyID = &testPolicyId
```

**Review by padamstx — APPROVED** [review]
LGTM

**Comment by swcolley** [comment]
retry Travis CI - Branch

**Pull request merged**